### DISCUSSION TITLE
istio observability and native support of OpenTelemetry (via envoy)

### DISCUSSION THREAD

**Discussion by itaykat** [teletrace/teletrace#52] [Ideas]
[Istio](https://istio.io/latest/docs/concepts/observability/) Service mesh generates telemetry data by itself, the context propagation  is currently [b3 based](https://github.com/openzipkin/b3-propagation), and traces are not transmitted using OTLP.
We should consider suggesting the standard of OpenTelemetry for service mesh.

In addition, I see some users complaining about the complexity of integrating a tracing backend (Zipkin, Jaeger, etc.), I see it as a potential collaboration that can be very fruitful:

- https://cloud-native.slack.com/archives/CJFCJHG4Q/p1636727327373000
- https://cloud-native.slack.com/archives/CJFCJHG4Q/p1660577248488249
- https://cloud-native.slack.com/archives/CJELJNX96/p1659017144707269
- https://cloud-native.slack.com/archives/CJELJNX96/p1638371264027600

Istio is a very mature CNCF project and the opportunity to enable cloud native open-source solution for its observability backend can bring real value to its users.
Please see a [comparison](https://ossinsight.io/analyze/istio/istio?vs=kiali%2Fkiali) between istio and [kiali](https://github.com/kiali/kiali) (which was mentioned in the first discussion)

An example for an [application](https://github.com/julianocosta89/opentelemetry-microservices-demo) that uses istio (this may be relevant for all service mash related project specifically and otel in general)

**Comment by itaykat** [comment]
We need to research the roadmap and future support of [envoy tracing on OpenTelemetry](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/observability/tracing) first

Related doc: https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/trace/v3/opentelemetry.proto.html

**Reply by itaykat** [reply]
We can definitely contribute here either way.
@osherv I'm interested hearing your take on it.

**Comment by roybh1** [comment]
@itaykat KNative is currently going through this transition to Opentelemetry. This could be a good reference later on
https://opentelemetry.io/blog/2022/knative/
https://github.com/knative/eventing/issues/3126
https://github.com/knative/pkg/issues/855
